extensions/eclipse-che-theia-workspace/src/browser/che-quick-open-workspace.ts

```diff
@@ -0,0 +1,229 @@
+                iconClass,
+                run: (mode: QuickOpenMode): boolean => {
+                    if (mode !== QuickOpenMode.OPEN) {
+                        return false;
+                    }
+
+                    if (this.isCurrentWorkspace(workspace)) {
+                        return true;
+                    }
+
+                    this.openWorkspace(workspace);
+
+                    return true;
+                },
+            }));
+        }
+
+        this.quickOpenService.open(this, {
+            placeholder: 'Type the name of the Che workspace you want to open',
+            fuzzyMatchLabel: true,
+            fuzzySort: false
+        });
+    }
+
+    onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+        acceptor(this.items);
+    }
+
+    async select(): Promise<void> {
+        this.items = [];
+
+        const token = await this.oAuthUtils.getUserToken();
+
+        if (!this.currentWorkspace) {
+            this.currentWorkspace = await this.cheApi.currentWorkspace();
+        }
+
+        if (!this.currentWorkspace.namespace) {
+            return;
+        }
+
+        const workspaces = await this.cheApi.getAllByNamespace(this.currentWorkspace.namespace, token);
+
+        workspaces.sort((a: cheApi.workspace.Workspace, b: cheApi.workspace.Workspace) => {
+            const updatedA: number = this.getWorkspaceModificationTime(a);
+            const updatedB: number = this.getWorkspaceModificationTime(b);
+
+            if (isNaN(updatedA) || isNaN(updatedB)) {
+                return 0;
+            } else {
+                return updatedB - updatedA;
+            }
+        });
+
+        await this.open(workspaces);
+    }
+
+    private getWorkspaceName(workspace: cheApi.workspace.Workspace): string | undefined {
+        if (workspace.config) {
+            return workspace.config.name;
+        } else if (workspace.devfile && workspace.devfile.metadata) {
+            return workspace.devfile.metadata.name;
+        }
+    }
+
+    private getWorkspaceStack(workspace: cheApi.workspace.Workspace): string | undefined {
+        if (workspace.attributes && workspace.attributes.stackName) {
+            return `Stack: ${workspace.attributes.stackName}`;
+        } else {
+            return 'Stack: Custom';
+        }
+    }
+
+    private getWorkspaceModificationTime(workspace: cheApi.workspace.Workspace): number {
+        if (workspace.attributes) {
+            if (workspace.attributes.updated) {
+                return parseInt(workspace.attributes.updated);
+            } else if (workspace.attributes.created) {
+                return parseInt(workspace.attributes.created);
+            }
+        }
+
+        return NaN;
+    }
+
+    private stopCurrentWorkspace(): Promise<boolean | undefined> {
+        class StopWorkspaceDialog extends AbstractDialog<boolean | undefined> {
```

Comment:
Maybe extract the class a separate file?
It can be reused for [`Close Workspace`](https://github.com/eclipse/che/issues/17238) feature.

Reply:
I thought about it, I'm going to extract this dialog into separate class during the next PR followed by this issue: https://github.com/eclipse/che/issues/17106#issue-631935000